rclpy/src/rclpy/_rclpy.c

```diff
@@ -121,7 +123,7 @@ rclpy_trigger_guard_condition(PyObject * Py_UNUSED(self), PyObject * args)
 {
   PyObject * pygc;
 
-  if (!PyArg_ParseTuple(args, "O", &pygc)) {
+  if (!PyArg_ParseTuple(args, "O", &pygc) || !PyCapsule_CheckExact(pygc)) {
```

Comment:
@sloretz I think this changed the behavior of the statement:
if PyArg_ParseTuple fails an error will be raised while if PyCapsule_CheckExact fails we return FALSE without raising. Is it expected or something we overlooked ?

Reply:
What was the behavior before? I think this returns false in both cases. The docs don't mention [`"O"` raising an exception, although `"O!"` does](https://docs.python.org/3/c-api/arg.html#other-objects).

Reply:
O! is only to check Object types, any failure to parse an argument will raise an exception if I read the [function description](https://docs.python.org/3/c-api/arg.html#c.PyArg_ParseTuple) correctly:
"Parse the parameters of a function that takes only positional parameters into local variables. Returns true on success; on failure, it returns false and raises the appropriate exception."

Reply:
Ah oops I missed that. Sounds like should be returning NULL on PyArg_ParseTuple failure and either raising an exception or returning False if the capsule check fails.

Reply:
I noticed that when preparing a PR to address https://github.com/ros2/rclpy/pull/120#pullrequestreview-63114812. I can fix this in the same PR. Would that conflict with your current work ?

Reply:
It would not conflict. I have a couple additions to this file, but no modifications.